src/x/tls/config_manager.go

```diff
@@ -76,28 +79,39 @@ func NewConfigManager(opts Options, instrumentOpts instrument.Options) ConfigMan
 		metrics:  newConfigManagerMetrics(scope),
 		options:  opts,
 		certPool: x509.NewCertPool(),
+		stopCh:   make(chan struct{}, 1),
 	}
-	go c.updateCertificates()
```

Comment:
this goroutine was leaking since we create a ConfigManager object for every new connection. Over time as new connections are created, we will keep accumulating this leak.